pwndbg/aglib/heap/jemalloc.py

```diff
@@ -0,0 +1,555 @@
+                        rtree_node_elm_s, node_address
+                    )
+                    node = pwndbg.gdblib.memory.pack_struct_into_dictionary(fetched_struct)
+
+                    if node["child"]["repr"] == 0:
+                        continue
+
+                    leaf0 = node["child"]["repr"]
+                    # print(hex(leaf0))
+
+                    # print("leaf0: ", leaf0)
+
+                    # level 1
+                    for j in range(max_subkeys):
+                        leaf_address = int(leaf0) + j * rtree_leaf_elm_s.sizeof
+                        # leaf = pwndbg.gdblib.memory.poi(rtree_leaf_elm_s, leaf)
+                        fetched_struct = pwndbg.gdblib.memory.get_typed_pointer_value(
+                            rtree_leaf_elm_s, leaf_address
+                        )
+                        leaf = pwndbg.gdblib.memory.pack_struct_into_dictionary(fetched_struct)
+
+                        if leaf["le_bits"]["repr"] == 0:
+                            continue
+
+                        # print("leaf: ", hex(leaf_address))
+
+                        # print(j, leaf)
+                        val = int(leaf["le_bits"]["repr"])
+
+                        if val == 0:
+                            # return None
+                            continue
+
+                        ls = (val << RTREE_NHIB) & ((2**64) - 1)
+                        ptr = ((ls >> RTREE_NHIB) >> 1) << 1
+
+                        if ptr == 0 or ptr == last_addr:
+                            continue
+
+                        last_addr = ptr
+
+                        extent = Extent(ptr)
+
+                        if extent.extent_address in extent_addresses:
+                            continue
+
+                        extent_addresses.append(extent.extent_address)
+
+                        # during initializations, addresses may get some alignment
+                        # lets check if size makes sense, otherwise do page alignment and check if again
+                        # TODO: better way to do this
+                        extent_tmp = extent
+                        if extent.size == 0:
+                            ptr = RTree.__alignment_addr2base(int(ptr))
+                            extent_tmp = Extent(ptr)
+                            if extent_tmp.size != 0:
+                                self._extents.append(extent_tmp)
+                                continue
+
+                        self._extents.append(extent_tmp)
+
+            except gdb.MemoryError:
+                pass
+
+        return self._extents
+
+
+class Arena:
+    """
+    Some notes:
+    - Huge allocation should not come from arena 0
+    """
+
+    def __init__(self, addr: int) -> None:
+        self._addr = addr
+
+        self._Value = pwndbg.gdblib.memory.fetch_struct_as_dictionary("arena_s", self._addr)
+
+        self._nbins = None
+        self._slabs = None
+
+    @property
+    def slabs(self):
```

Comment:
What is this property for? It seems to be unused across the code and it seems to be broken: it uses `self._bins` which is not defined in the `Arena` class.

cc: @jetchirag

Reply:
@disconnect3d I left this for future extension as I started working on this project with this property but it's not needed right now. I'll remove Arenas.